aws/resource_aws_cloudwatch_event_bus_test.go

```diff
@@ -0,0 +1,184 @@
+		if err != nil {
+			if testSweepSkipSweepError(err) {
+				log.Printf("[WARN] Skipping CloudWatch Event Bus sweep for %s: %s", region, err)
+				return nil
+			}
+			return fmt.Errorf("Error retrieving CloudWatch Event Buses: %s", err)
+		}
+
+		if len(output.EventBuses) == 0 {
+			log.Print("[DEBUG] No CloudWatch Event Buses to sweep")
+			return nil
+		}
+
+		for _, eventBus := range output.EventBuses {
+			name := aws.StringValue(eventBus.Name)
+			if name == "default" {
+				continue
+			}
+
+			log.Printf("[INFO] Deleting CloudWatch Event Bus %s", name)
+			_, err := conn.DeleteEventBus(&events.DeleteEventBusInput{
+				Name: aws.String(name),
+			})
+			if err != nil {
+				return fmt.Errorf("Error deleting CloudWatch Event Bus %s: %s", name, err)
+			}
+		}
+
+		if output.NextToken == nil {
+			break
+		}
+		input.NextToken = output.NextToken
+	}
+
+	return nil
+}
+
+func TestAccAWSCloudWatchEventBus_basic(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloudWatchEventBusDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudWatchEventBusConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudWatchEventBusExists("aws_cloudwatch_event_bus.foo"),
+					resource.TestCheckResourceAttr("aws_cloudwatch_event_bus.foo", "name", "tf-acc-cw-event-bus"),
+				),
+			},
+			{
+				Config: testAccAWSCloudWatchEventBusConfigModified,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudWatchEventBusExists("aws_cloudwatch_event_bus.foo"),
+					resource.TestCheckResourceAttr("aws_cloudwatch_event_bus.foo", "name", "tf-acc-cw-event-bus-mod"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSCloudWatchEventBus_withRule(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloudWatchEventBusDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudWatchEventBusConfigRule,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudWatchEventBusExists("aws_cloudwatch_event_bus.foo"),
+					resource.TestCheckResourceAttr("aws_cloudwatch_event_bus.foo", "name", "tf-acc-cw-event-bus-with-rule"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSCloudWatchEventBusDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).cloudwatcheventsconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_cloudwatch_event_bus" {
+			continue
+		}
+
+		params := events.DescribeEventBusInput{
+			Name: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.DescribeEventBus(&params)
+
+		if err == nil {
+			return fmt.Errorf("CloudWatch Event Bus %q still exists: %s",
+				rs.Primary.ID, resp)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckCloudWatchEventBusExists(n string) resource.TestCheckFunc {
```

Comment:
May want to add a 2nd parameter `v *events.DescribeEventBusOutput` and set it (after the `resp == nil` check):

```go
*v = *resp
```